pandas/core/algorithms.py

```diff
@@ -1101,6 +1102,10 @@ def __init__(self, obj, n: int, keep: str):
         if self.keep not in ("first", "last", "all"):
             raise ValueError('keep must be either "first", "last" or "all"')
 
+    @abc.abstractmethod
+    def compute(self, method: str) -> FrameOrSeriesUnion:
```

Comment:
if you make SelectN generic and use `FrameOrSeries` typevar you can also get the return types for nlargest and nsmallest sorted. follow-on OK